File: solutions/python/task_002.py
def descending_order(num):
    n = []
    if num == 0:
        return 0
    while num != 0:
        n.append(num%10)
        num = num // 10
    n.sort(reverse=True)
    to_return = 0
    ten = 10**(len(n)-1)
    for i in n:
        to_return += (ten*i)
        ten = ten // 10
    
    
    return to_return

assert descending_order(0) == 0
assert descending_order(15) == 51
assert descending_order(123456789) == 987654321
print('Test pass.')

File: solutions/cpp/task_002.cpp
#include <iostream>
#include <cassert>
#include <vector>
#include <algorithm>
#include <cmath>

int descending_order(int n)
{
    if(n==0)
        return 0;
    std::vector<int> digits;
    while(n>0){
        digits.push_back(n%10);
        n = n / 10;
    }

    std::sort(digits.rbegin(), digits.rend());
    int to_return = 0;
    int m = pow(10, digits.size()-1);
    for(auto it = digits.begin(); it != digits.end(); it++){
        to_return += (*it*m);
        m = m / 10;
    }

    return to_return;
    
}

int main(){
    std::cout << "Testing number: " << 0 << "\t result: " << descending_order(0) << "\n";
    std::cout << "Testing number: " << 15 << "\t result: " << descending_order(15) << "\n";
    std::cout << "Testing number: " << 123456789 << "\t result: " << descending_order(123456789) << "\n";

    assert(descending_order(0) == 0);
    assert(descending_order(15) == 51);
    assert(descending_order(123456789) == 987654321);
    
    std::cout << "Test pass." << std::endl;
}